src/frontend/XAPI/VideoXAPIStatement.ts

```diff
@@ -66,28 +111,6 @@ export class VideoXAPIStatement implements VideoXAPIStatementInterface {
 
         return acc;
       }, [])
-      // sort segments (numerically)
```

Comment:
Did you try removing the `filter`, `map` and `reduce` before ?

Reply:
With the current code structure, it is not possible as the `getProgress` input is a string and the _compute progression_ step input is an array

Reply:
The `non complete segments` are already removed in the `mergeSegments`, IMO this filter can be removed, the tests are still green when I remove it. Or the tests are wrong 😁

Reply:
Indeed, the remaining filter was useless. I removed it